### PR TITLE
RCF detector utilising modelservice

### PR DESCRIPTION
**Pull request by tkamenov-expedia**
Fully operational RCF detector based on data obtained from modelservice.

### CODE REVIEW

File: anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/rcf/ModelParameters.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import lombok.experimental.Accessors;
+
+/**
+ * @author Tatjana Kamenov
+ */
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Accessors(chain = true)
+public class ModelParameters {
```
Comment (djsutho) [inline anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/rcf/ModelParameters.java]: To be consistent with the other models should the be RcfParams e.g. https://github.com/ExpediaDotCom/adaptive-alerting/blob/a761ad93d51f2ab8f7ba0e98359a5eb1bc0c14b2/anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/ewma/EwmaParams.java

File: anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/rcf/RandomCutForestAnomalyDetector.java [hunk 2]
```diff
@@ -109,29 +132,35 @@ public AnomalyResult classify(MetricData metricData) {
...
+        // scores back by simple updates to this class. [TK]
         final String shingleBody = this.shingle.toCsv().get();
         final Optional<ByteBuffer> bodyBuffer =
                 Optional.of(ByteBuffer.wrap(shingleBody.getBytes(StandardCharsets.UTF_8)));
 
         if (bodyBuffer.isPresent()) {
-            invokeEndpointRequest.setBody(bodyBuffer.get());
-            invokeEndpointRequest.setEndpointName(ENDPOINT);
-            invokeEndpointRequest.setAccept(APPLICATION_JSON_ACCEPT);
+            this.invokeEndpointRequest.setBody(bodyBuffer.get());
```
Comment (djsutho) [inline anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/rcf/RandomCutForestAnomalyDetector.java]: Is it ever possible for this to be called concurrently? Worried about reusing the same request.

File: anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/rcf/RandomCutForestAnomalyDetectorFactory.java [hunk 3]
```diff
@@ -16,28 +16,67 @@
...
-        
-        // TODO Return different models for different metrics. [WLW]
-        return new RandomCutForestAnomalyDetector(uuid);
+
+        final Resources<ModelResource> models = modelServiceConnector.findModels(uuid);
+        final Collection<ModelResource> modelResources = models.getContent();
+
+        List<ModelResource> modelResourceList = new ArrayList<>(modelResources);
+        if (modelResourceList.isEmpty()) {
+            log.error("There is no RCF model associated with uuid: {}", uuid);
```
Comment (djsutho) [inline anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/rcf/RandomCutForestAnomalyDetectorFactory.java]: Should this throw? Next call `modelResourceList.get(0)` will fail.

File: anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/rcf/ModelParameters.java (hunk 1 above)
Comment (tkamenov-expedia) [reply anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/rcf/ModelParameters.java]: These are optional after-training parameters specific to rcf so I did not want to make confusion.

**Pull request merged**